app/controllers/group_assignments_controller.rb

```diff
@@ -35,7 +35,8 @@ def edit
 
   def update
     if @group_assignment.update_attributes(update_group_assignment_params)
-      flash[:success] = "Assignment \"#{@group_assignment.title}\" updated"
+      AssignmentVisibilityJob.perform_later(@group_assignment)
```

Comment:
What do you think about checking to see if the privacy setting was changed before kicking off the background job?

Otherwise if someone changes the title this will iterate through all of the repositories which is an expensive process.